doc/scripting_tr2_tr3.asc

```diff
@@ -14,98 +14,99 @@ of flexibility.
-uint16_t CutscenePathStringOffsets[NumCutscenes+1];       //Relative offset to cutscene path strings
-StringArray CutscenePathStrings[NumCutscenes];            //Usually Encrypted
-
-uint16_t SequenceOffsets[NumLevels+2]; //Relative offset to sequence info (maybe+2 because title needs sequence +numtitles?)
-SequenceInfo Sequences[NumLevels+1];   //Sequence info see explanation below (SIZE is dependant on first opcode)
+// 3. If Flags & UseXor true each character (except null-terminator) must be ^ XorKey to decrypt the string.
+
+uint16_t LevelStringOffsets[NumLevels+1];                 // Relative offset to level name strings
+StringArray LevelStrings[NumLevels];                      // Usually Encrypted
+uint16_t ChapterScreenStringOffsets[NumChapterScreens+1]; // Relative offset to chapter screen strings
+StringArray ChapterScreenStrings[NumChapterScreens];      // Usually Encrypted
+uint16_t TitleStringOffsets[NumTitles+1];                 // Relative offset to title strings
+StringArray TitleStrings[NumTitles];                      // Usually Encrypted
+uint16_t FMVStringOffsets[NumFMVs+1];                     // Relative offset to fmv path strings
+StringArray FMVStrings[NumFMVs];                          // Usually Encrypted
+uint16_t LevelPathStringOffsets[NumLevels+1];             // Relative offset to level path strings
+StringArray LevelPathStrings[NumLevels];                  // Usually Encrypted
+uint16_t CutscenePathStringOffsets[NumCutscenes+1];       // Relative offset to cutscene path strings
+StringArray CutscenePathStrings[NumCutscenes];            // Usually Encrypted
+
+uint16_t SequenceOffsets[NumLevels + 1]; // Relative offset to sequence info (the +1 is because the first one is the FrontEnd sequence, for when the game starts)
+uint16_t SequenceNumBytes; // Size of SequenceOffsets in bytes
+SequenceInfo Sequences[NumLevels + 1];   // Sequence info see explanation below (SIZE is dependant on first opcode)
 
 uint16_t DemoLevelIDs[NumDemoLevels];
 
 #if PSX //PSX Only
-    PSXFMVInfo[NumFMVs];// Two uint32_t, flag? then length/lba related?
+    PSXFMVInfo[NumFMVs];
 #endif
 
-int16_t NumGameStrings;
-uint16_t GameStringOffsets[NumGameStrings+1];   //Relative offset to game strings
-StringArray GameStrings[NumGameStrings];        //Usually Encrypted
+uint16_t NumGameStrings;
+uint16_t GameStringOffsets[NumGameStrings+1];   // Relative offset to game strings
+StringArray GameStrings[NumGameStrings];        // Usually Encrypted
 
 #if PSX // If PSX
-    uint16_t PSXStringOffsets[80+1];//Fixed at 80?
-    StringArray PSXStrings[80+1];//Fixed at 80? Usually Encrypted
+    uint16_t PSXStringOffsets[size + 1]; // size is 79 for the TR2 beta, 80 for all other versions
+    StringArray PSXStrings[size];
 #else // PC
-    uint16_t PCStringOffsets[80+1];//Fixed at 80?
-    StringArray PCStrings[80+1];//Fixed at 80? Usually Encrypted
+    uint16_t PCStringOffsets[41 + 1];
```

Comment:
I'm fairly certain this should not be 41+1. Size is possible to calculate with hacks as well (unrelated).